packages/common/src/urls/get-portal-url.ts

```diff
@@ -28,7 +25,11 @@ import { IHubRequestOptions } from "../types";
  * @returns The portal base URL, defaults to `https://www.arcgis.com`
  */
 export function getPortalUrl(
-  urlOrObject?: string | IPortal | IHubRequestOptions | IRequestOptions
+  urlOrObject?:
+    | string
+    | IHubRequestOptionsPortalSelf
```

Comment:
@tomwayson 

`IHubRequestOptionsPortalSelf` extends `IPortal` and only adds an _optional_ `user` property, therefore this change from `string | IPortal | IHubRequestOptions | IRequestOptions` to `string | IHubRequestOptionsPortalSelf | IHubRequestOptions | IRequestOptions` should be transparent to our users and a non-breaking change, but please check me on this. I'm also happy to change this to `string | IPortal | IHubRequestOptionsPortalSelf | IHubRequestOptions | IRequestOptions`, too, if you think that makes more sense.

Reply:
I'm not 100% sure that will be non-breaking b/c the interface name changed... if you just add an optional prop to an interface, that's non-breaking... this will be fine for JS clients, but I _suspect_ typescript libs will have issues (i.e. Solution.js)